core/src/main/java/org/frankframework/core/ISecurityHandler.java

```diff
@@ -27,6 +28,10 @@
  */
 public interface ISecurityHandler {
 
+	boolean isUserInAnyRole(List<String> roles, PipeLineSession session) throws NotImplementedException;
+
+	String inWhichRoleIsUser(List<String> roles, PipeLineSession session) throws NotImplementedException;
```

Comment:
If this becomes an actual requirement in the future, it is simply added by providing a new method.

Reply:
The interface can provide a method to return all user roles, I think that can be generally useful and that can then be used in the `IsUserInRolePipe`.

Reply:
Hmm, could you explain why this is considered as 'not the right approach' ?

In my opinion, by implementing the method in `IsUserInRolePipe` , it's not really re-usable..  what about a helper class which exposes these methods and takes an `ISecurityHandler` and the roles to check?

Reply:
You're imposing limitations on the interface which only apply to the specific code of the `IsUserInRolePipe`. The underlying data that the interface encapsulates, does not have these limitations.
That's why I do not think the interface should return only the first role.

Either the interface should return the set of all roles (doesn't need arguments for that), or the interface should return a set of all overlapping roles: `Set<String> findMatchingRoles(Collection<String> roles)`.

Since the `IsUserInRolePipe` is the one that is limited to only handling a single role, that pipe should sort that set alphabetically and pick the first entry.

Reply:
PS: About helper class, I think that is not needed. I think that methods which work the same for all implementations can be added as default methods to the interface, or in a base-class as is done now.